src/Swift.Bindings/src/DemangledSymbols.cs

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using BindingsGeneration;
+using Xamarin;
+
+public readonly record struct DemangledSymbols
+{
+    public Dictionary<(NamedTypeSpec, NamedTypeSpec), string> ProtocolConformanceDescriptors { get; init; }
+
+    public DemangledSymbols(Dictionary<(NamedTypeSpec, NamedTypeSpec), string> descriptors)
+    {
+        ProtocolConformanceDescriptors = descriptors;
+    }
+}
+
+public sealed class DemangledSymbolsRegister
```

Comment:
fui @matouskozak As mentioned in the description this is just a temporary solution to getting symbols until https://github.com/dotnet/runtimelab/issues/2918 is implemented. I tried to put all the logic here so it can be easily removed. I initially thought we can just synthesise the symbols for now, but it is more complicated than I thought.

This is not gonna work for Apple shipped modules which dont have easily accessible dylib (CryptoKit or Foundation) as Demangler will fail to resolve the library. The error is not critical. It will result in `GetProtocolConformance` methods not being populated.